### PR TITLE
descriptor schema: add missing data and artifactType definitions

### PR DESCRIPTION
**Pull request by vbatts**
Fixes: #1019

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

**Comment by vbatts** [comment]
I'm honestly not sure I've done the base64 definition correct. It's basically only a `string`. 
https://cswr.github.io/JsonSchema/spec/grammar/
We'd have to find a failing test of invalid data to refine it. I was lacking creativity to make that.

File: schema/descriptor_test.go [hunk 1]
```diff
@@ -232,6 +232,46 @@ func TestDescriptor(t *testing.T) {
...
+			fail: true,
+		},
+
+		// expected success: data field is present and has base64 content
+		{
+			descriptor: `
+		{
+			"mediaType": "text/plain",
+			"size": 4,
+			"data": "ZmFydHMK",
```
Comment (sudo-bmitch) [inline schema/descriptor_test.go:268]: I'd prefer we used a different example value here.

Comment (vbatts) [reply schema/descriptor_test.go:268]: dang, @sudo-bmitch

Comment (vbatts) [reply schema/descriptor_test.go:268]: so official

**Comment by vbatts** [comment]
updated PTAL

File: schema/defs.json [hunk 2]
```diff
@@ -61,6 +61,12 @@
         }
       ]
     },
+    "base64": {
+      "type": "string",
+      "media": {
+        "binaryEncoding": "base64"
```
Comment (tianon) [inline schema/defs.json:67]: Shouldn't this be `contentEncoding`?
(http://json-schema.org/draft/2020-12/json-schema-validation.html#name-contentencoding)

**Comment by vbatts** [comment]
updated PTAL

**Review by sajayantony — APPROVED** [review]

File: schema/defs.json [hunk 3]
```diff
@@ -61,6 +61,12 @@
         }
       ]
     },
+    "base64": {
+      "type": "string",
+      "media": {
+        "contentEncoding": "base64"
```
Comment (sudo-bmitch) [inline schema/defs.json:67]: Draft 4 changed this to "binaryEncoding": https://datatracker.ietf.org/doc/html/draft-luff-json-hyper-schema-00#section-4.3.1.1
```suggestion
        "binaryEncoding": "base64"
```

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]

File: schema/defs.json (hunk 3 above)
Comment (tianon) [reply schema/defs.json:67]: See https://github.com/opencontainers/image-spec/pull/1022#discussion_r1104938172, which links to a 2020 draft (newer than draft 4, as far as I can find) that disagrees :sweat_smile:

**Comment by sudo-bmitch** [comment]
Here's a test for invalid base64 (I just removed the required `==` from the end of the other example):

```
		{
			descriptor: `
		{
			"mediaType": "text/plain",
			"size": 34,
			"data": "aHR0cHM6Ly9naXRodWIuY29tL29wZW5jb250YWluZXJzCg",
			"digest": "sha256:2690af59371e9eca9453dc29882643f46e5ca47ec2862bd517b5e17351325153"
		}
		`,
			fail: true,
		},
```

File: schema/defs.json (hunk 2 above)
Comment (sudo-bmitch) [reply schema/defs.json]: Not in draft 4 which we are using. See https://json-schema.org/specification-links.html which links to https://datatracker.ietf.org/doc/html/draft-luff-json-hyper-schema-00#section-4.3.1.1

Comment (tianon) [reply schema/defs.json]: Doh, sorry @vbatts :upside_down_face: :heart:

File: schema/defs.json (hunk 3 above)
Comment (sudo-bmitch) [reply schema/defs.json:67]: I wouldn't be opposed to updating to the newer drafts, but I think that's a bigger lift for us (double checking for any other changes, and updating the draft link in the various parts of the spec).

Comment (sudo-bmitch) [reply schema/defs.json:67]: I'm being dragged into learning the json-spec stuff kicking and screaming after I changed all the `$schema` lines to https incorrectly. See #931 for fixing that mistake.

**Comment by vbatts** [comment]
updated PTAL

**Review by sudo-bmitch — APPROVED** [review]
LGTM. Thanks!

**Review by sajayantony — APPROVED** [review]

**Pull request merged**